### PR TITLE
ci: fix host name resolution

### PR DESCRIPTION
**Pull request by nrainer-materialize**
More a temporary thing to understand what we get exactly in this case: https://materializeinc.slack.com/archives/C057X0NG91B/p1721914973133919

### CODE REVIEW

File: bin/ci-annotate-errors [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
 HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password-env-var "PRODUCTION_ANALYTICS_APP_PASSWORD")
 
+echo "Returned hostname is:"
+echo "$HOSTNAME"
+
```
Comment (def-) [inline bin/ci-annotate-errors:19]:
```suggestion
# Only take the last line, previous content can be retries when docker pull has problems
HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password-env-var "PRODUCTION_ANALYTICS_APP_PASSWORD" | tail -n1)
echo "Using hostname: $HOSTNAME"
```

**Review by def- — CHANGES_REQUESTED** [review]
I thought we know it's retrying already? This happens when the `docker pull` has a failure:
https://github.com/MaterializeInc/materialize/blob/73b543cbb3abff810af9fcfc9205e952686f0d9a/misc/python/materialize/mzbuild.py#L649-L671

**Comment by nrainer-materialize** [comment]
Hm, no, the last line is also not ok. Changed it to take the second last line because "there is a further line containing the mzcompose result summary".

File: bin/ci-annotate-errors [hunk 2]
```diff
@@ -12,7 +12,8 @@
 # ci-annotate-errors - Detect errors in junit xml as well as log files during
 # CI and find associated open GitHub issues in Materialize repository.
 
-HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password-env-var "PRODUCTION_ANALYTICS_APP_PASSWORD")
+# Only take last two lines, previous content can be retries when docker pull has problems
+HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password-env-var "PRODUCTION_ANALYTICS_APP_PASSWORD" | tail -n2)
```
Comment (def-) [inline bin/ci-annotate-errors:16]: Doesn't this contain a newline then? How can the hostname have a newline?
```suggestion
HOSTNAME=$(exec "$(dirname "$0")"/ci-builder run stable "$(dirname "$0")"/mzcompose --mz-quiet --find get-cloud-hostname run default --app-password-env-var "PRODUCTION_ANALYTICS_APP_PASSWORD" | tail -n2 | head -n1)
```

**Review by def- — CHANGES_REQUESTED** [review]

File: bin/ci-annotate-errors (hunk 2 above)
Comment (nrainer-materialize) [reply bin/ci-annotate-errors:16]: This is done in line 18 already...

Comment (nrainer-materialize) [reply bin/ci-annotate-errors]: I slightly refactored it now.

**Review by def- — APPROVED** [review]

**Pull request merged**